### PR TITLE
feat(onboarding): Add onboarding welcome page experiment

### PR DESCRIPTION
**Pull request by scefali**
image
image
image
image

### CODE REVIEW

File: static/app/data/experimentConfig.tsx [hunk 1]
```diff
@@ -14,31 +14,19 @@ export const unassignedValue = -1;
...
-    assignments: ['none', 'on_signup', 'on_first_event'],
-  },
-  {
-    key: 'SubscriptionMemberViewExperiment',
-    type: ExperimentType.Organization,
-    parameter: 'exposed',
-    assignments: [0, 1],
-  },
-  {
-    key: 'ContactSalesOnCheckoutExperiment',
```
Comment (evanpurkhiser) [inline static/app/data/experimentConfig.tsx:29]: Would be nice to just remove these in a separate PR to get it in now

Comment (evanpurkhiser) [inline static/app/data/experimentConfig.tsx:29]: Would be nice to just remove these in a separate PR to get it in now

File: static/app/components/demoSandboxButton.tsx [hunk 2]
```diff
@@ -32,6 +32,11 @@ interface DemoSandboxButtonProps extends ButtonPropsWithoutAriaLabel {
    * Which project we should link to in the sandbox
    */
   projectSlug?: 'react' | 'python' | 'ios' | 'android' | 'react-native';
+
+  /**
+   * Where is the component being used
+   */
+  source?: string;
```
Comment (evanpurkhiser) [inline static/app/components/demoSandboxButton.tsx:39]: Can also prob happen in a sep PR to get it in now

Comment (evanpurkhiser) [inline static/app/components/demoSandboxButton.tsx:39]: Can also prob happen in a sep PR to get it in now

File: static/app/views/onboarding/onboardingController.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
```
Comment (evanpurkhiser) [inline static/app/views/onboarding/onboardingController.tsx:1]: You probably don't need this

Comment (evanpurkhiser) [inline static/app/views/onboarding/onboardingController.tsx:1]: You probably don't need this

File: static/app/views/onboarding/onboardingController.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    logExperiment({
+      key: 'TargetedOnboardingWelcomePageExperiment',
+      organization: rest.organization,
+    });
+  }, []);
+  const {
+    params: {step},
+  } = rest;
+  if (step === 'welcome' && experimentAssignment) {
+    return <TargetedOnboarding />;
```
Comment (evanpurkhiser) [inline static/app/views/onboarding/onboardingController.tsx:26]:
```suggestion
  if (rest.params.step === 'welcome' && experimentAssignment) {
    return <TargetedOnboarding />;
```

Comment (evanpurkhiser) [inline static/app/views/onboarding/onboardingController.tsx:26]:
```suggestion
  if (rest.params.step === 'welcome' && experimentAssignment) {
    return <TargetedOnboarding />;
```

File: static/app/views/onboarding/targetedOnboarding/components/welcomeBackground.tsx [hunk 5]
```diff
@@ -0,0 +1,320 @@
+import * as React from 'react';
+import {keyframes} from '@emotion/react';
+import styled from '@emotion/styled';
+import {motion} from 'framer-motion';
+import random from 'lodash/random';
+
+import testableTransition from 'sentry/utils/testableTransition';
+
+// Using framer motion for this loop for the nice physics
```
Comment (evanpurkhiser) [inline static/app/views/onboarding/targetedOnboarding/components/welcomeBackground.tsx:9]: Do we need this comment really?

Comment (evanpurkhiser) [inline static/app/views/onboarding/targetedOnboarding/components/welcomeBackground.tsx:9]: Do we need this comment really?

File: static/app/views/onboarding/targetedOnboarding/onboarding.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+    return (
+      <Container>
+        <AnimatePresence exitBeforeEnter onExitComplete={updateCornerVariant}>
+          <TargetedOnboardingWelcome />
+        </AnimatePresence>
+        <PageCorners animateVariant={cornerVariantControl} />
+      </Container>
+    );
+  };
```
Comment (evanpurkhiser) [inline static/app/views/onboarding/targetedOnboarding/onboarding.tsx:35]: This only needed to be in it's own component before bc the `useAnimation` couldn't be used in a classic React.Component class.

Comment (evanpurkhiser) [inline static/app/views/onboarding/targetedOnboarding/onboarding.tsx:35]: This only needed to be in it's own component before bc the `useAnimation` couldn't be used in a classic React.Component class.

**Review by evanpurkhiser — APPROVED** [review]
looks fine to me for an experiment

File: static/app/views/onboarding/targetedOnboarding/components/welcomeBackground.tsx (hunk 5 above)
Comment (Neo-Zhixing) [reply static/app/views/onboarding/targetedOnboarding/components/welcomeBackground.tsx:9]: Well this is copy pasted from your version https://github.com/getsentry/sentry/blob/ba15b6285fd8d190e249f3acc3fe42927d622f21/static/app/views/onboarding/components/welcomeBackground.tsx#L9

Comment (evanpurkhiser) [reply static/app/views/onboarding/targetedOnboarding/components/welcomeBackground.tsx:9]: hahaha it was me :) ok nvm

File: static/app/views/onboarding/onboardingController.tsx [hunk 7]
```diff
@@ -12,7 +13,12 @@ type Props = Omit<React.ComponentPropsWithoutRef<typeof Onboarding>, 'projects'>
 };
 
 function OnboardingController({experimentAssignment, ...rest}: Props) {
-  // TODO: call logExperiment
+  React.useEffect(() => {
```
Comment (scefali) [inline static/app/views/onboarding/onboardingController.tsx]: @Neo-Zhixing Can we remove this until after we've tested the new flow out on prod? This was supposed to be handled in a future PR: https://getsentry.atlassian.net/browse/GRW-473

**Pull request merged**